wolfcrypt/src/asn.c

```diff
@@ -23198,7 +23198,7 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
 #endif
             {
                 WOLFSSL_ERROR_VERBOSE(ASN_NO_SIGNER_E);
-                return ASN_NO_SIGNER_E;
+                ret = ASN_NO_SIGNER_E;
```

Comment:
Should the `return ASN_SELF_SIGNED_E;` above also be changed?

Reply:
It definitely should. Thanks for catching that

Reply:
There are other `return ret` cases above too. Please review if those need changed also.

Reply:
I can change them for consistencies sake but I believe the `else` case returning `ASN_SELF_SIGNED_E` or `ASN_NO_SIGNER_E` were the only points that yielded undesired behavior

Reply:
Great, then only change these two. Can you document what the undesirable behavior is here?

Reply:
Originally I thought this was a matter of error prioritization but this PR is just wrong. Customer was using verify callbacks with the cert manager to override date errors when loading a CA but I now realize that adding a CA and simply overwriting the return value regardless of the result is invalid. To get what they want to do to work, some logic will on the verify  [here](https://github.com/wolfSSL/wolfssl/blob/210fff55695141ca8797b5644a8215aab49f2e5a/src/ssl.c#L5740) will have to be added based on if `cm->verifyCallback` is set. I've instead recommended the use of `wolfSSL_CertManagerLoadCABuffer_ex()`